jscomp/others/js_json.ml

```diff
@@ -61,6 +61,12 @@ let test (type a) (x : 'a) (v : a kind) : bool =
   | Array -> Js_array.isArray x 
   | Object -> (Obj.magic x) != Js.null && Js.typeof x = "object" && not (Js_array.isArray x )
 
+let stringOfJson json = 
+  let ty, x = reifyType json in 
```

Comment:
this API is good to me, for the implementation, you can just use `typeof` and magic which should be fine